src/components/BasicChart.jsx

```diff
@@ -0,0 +1,913 @@
+                                },
+                            },
+                        ]}
+                    />
+                </div>
+                {config.brush ?
+                    <div
+                        style={{ width: '80%', height: 40, display: 'inline', float: 'left', right: 10 }}
+                    >
+                        <div
+                            style={{ width: '10%', display: 'inline', float: 'left', left: 20 }}
+                        >
+                            <button onClick={() => {
+                                this.setState({ xDomain: this.xRange });
+                            }}
+                            >Reset
+                            </button>
+                        </div>
+                        <div
+                            style={{ width: '90%', display: 'inline', float: 'right' }}
+                        >
+                            <Range
+                                max={xScale === 'time' ? this.xRange[1].getDate() : this.xRange[1]}
+                                min={xScale === 'time' ? this.xRange[0].getDate() : this.xRange[0]}
+                                defaultValue={xScale === 'time' ?
+                                    [this.xRange[0].getDate(), this.xRange[1].getDate()] :
+                                    [this.xRange[0], this.xRange[1]]
+                                }
+                                value={xScale === 'time' ?
+                                    [this.state.xDomain[0].getDate(), this.state.xDomain[1].getDate()] :
+                                    this.state.xDomain}
+                                onChange={(d) => {
+                                    this.setState({
+                                        xDomain: d,
+                                    });
+                                }}
+                            />
+                        </div>
+                    </div> : null
+                }
+            </div >
+
+        );
+    }
+}
+
+BasicCharts.defaultProps = {
+    width: 800,
+    height: 450,
+    onClick: null,
+};
+
+BasicCharts.propTypes = {
+    width: PropTypes.number,
+    height: PropTypes.number,
+    onClick: PropTypes.func,
+    config: PropTypes.shape({
+        x: PropTypes.string,
+        charts: PropTypes.arrayOf(PropTypes.shape({
+            type: PropTypes.string.isRequired,
+            y: PropTypes.string.isRequired,
+            fill: PropTypes.string,
+            color: PropTypes.string,
+            colorScale: PropTypes.arrayOf(PropTypes.string),
+            colorDomain: PropTypes.arrayOf(PropTypes.string),
+            mode: PropTypes.string,
+        })),
+        tickLabelColor: PropTypes.string,
+        legendTitleColor: PropTypes.string,
+        legendTextColor: PropTypes.string,
+        axisColor: PropTypes.string,
+        height: PropTypes.number,
+        width: PropTypes.number,
+        maxLength: PropTypes.number,
+    }).isRequired,
```

Comment:
This class is very long. Try to break into more classes depending on the functionality

Reply:
Did a refactoring to the class